Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -122,44 +125,79 @@ import Foundation
         //   - If cachedConfiguration is not present, fetch it and cache the successful response
         //   - If fetching fails, return error
 
-        var configPath: String = "v1/configuration"
-
-        if let clientToken {
-            configPath = clientToken.configURL.absoluteString
-        }
-        
-        guard let authorization = clientToken?.authorizationFingerprint ?? tokenizationKey else {
-            completion(nil, BTAPIClientError.configurationUnavailable)
-            return
-        }
-        
-        if let cachedConfig = try? ConfigurationCache.shared.getFromCache(authorization: authorization) {
-            setupHTTPCredentials(cachedConfig)
-            completion(cachedConfig, nil)
-            return
-        }
-
-        configurationHTTP?.get(configPath, parameters: BTConfigurationRequest()) { [weak self] body, response, error in
+        fetchConfigQueue.async { [weak self] in
             guard let self else {
-                completion(nil, BTAPIClientError.deallocated)
+                DispatchQueue.main.async {
+                    completion(nil, BTAPIClientError.deallocated)
+                }
                 return
             }
 
-            if error != nil {
-                completion(nil, error)
-                return
-            } else if response?.statusCode != 200 || body == nil {
-                completion(nil, BTAPIClientError.configurationUnavailable)
+            var configPath: String = "v1/configuration"
+
+            if let clientToken {
+                configPath = clientToken.configURL.absoluteString
+            }
+
+            guard let authorization = clientToken?.authorizationFingerprint ?? tokenizationKey else {
+                DispatchQueue.main.async {
+                    completion(nil, BTAPIClientError.configurationUnavailable)
+                }
                 return
-            } else {
-                let configuration = BTConfiguration(json: body)
+            }
 
-                setupHTTPCredentials(configuration)
-                try? ConfigurationCache.shared.putInCache(authorization: authorization, configuration: configuration)
-                
-                completion(configuration, nil)
+            if let cachedConfig = try? ConfigurationCache.shared.getFromCache(authorization: authorization) {
+                setupHTTPCredentials(cachedConfig)
+                DispatchQueue.main.async {
```

Comment:
I added doing the completions on main because get call from BTHTTP returned results via dispatchQueue which was defaulted to main and I didn't see any calls overriding that. I noticed this because our demo setup for ApplePay had a crash because it was expecting results via main. I thought keeping this consistent with completion from get call completion block below would cause unexpected behavior in merchant app. https://github.com/braintree/braintree_ios/blob/da4c1056d4ae199917fb355f8c3ebe7118e3f03c/Sources/BraintreeCore/BTHTTP.swift#L311